pysindy/optimizers/trapping_sr3.py

```diff
@@ -0,0 +1,516 @@
+        function.
+
+    history_ : list
+        History of sparse coefficients. ``history_[k]`` contains the
+        sparse coefficients (v in the optimization objective function)
+        at iteration k.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from scipy.integrate import odeint
+    >>> from pysindy import SINDy
+    >>> from pysindy.optimizers import trappingSR3
+    >>> lorenz = lambda z,t : [10*(z[1] - z[0]),
+    >>>                        z[0]*(28 - z[2]) - z[1],
+    >>>                        z[0]*z[1] - 8/3*z[2]]
+    >>> t = np.arange(0,2,.002)
+    >>> x = odeint(lorenz, [-8,8,27], t)
+    >>> opt = trappingSR3(threshold=0.1)
+    >>> model = SINDy(optimizer=opt)
+    >>> model.fit(x, t=t[1]-t[0])
+    >>> model.print()
+    x0' = -10.004 1 + 10.004 x0
+    x1' = 27.994 1 + -0.993 x0 + -1.000 1 x1
+    x2' = -2.662 x1 + 1.000 1 x0
+    """
+
+    def __init__(
+        self,
+        w_evo=True,
+        threshold=0.1,
+        eps_solver=1.0e-7,
+        relax_optim=True,
+        eta=1.0e20,
+        alpha_A=1.0e20,
+        alpha_m=5e19,
+        gamma=-0.1,
+        tol=1e-5,
+        mtol=1e-5,
+        thresholder="l1",
+        max_iter=30,
+        accel=False,
+        normalize=False,
+        fit_intercept=False,
+        copy_X=True,
+        initial_guess=None,
+        m0=None,
+        A0=None,
+        PL=None,
+        PQ=None,
+        thresholds=None,
+        objective_history=None,
+        constraint_lhs=None,
+        constraint_rhs=None,
+        constraint_order="target",
+    ):
+        super(trappingSR3, self).__init__(
+            max_iter=max_iter,
+            initial_guess=initial_guess,
+            normalize=normalize,
+            fit_intercept=fit_intercept,
+            copy_X=copy_X,
+        )
+
+        if threshold < 0:
+            raise ValueError("threshold cannot be negative")
+        if eta <= 0:
+            raise ValueError("eta must be positive")
+        if alpha_m < 0:
+            raise ValueError("alpha_m must be positive")
+        if alpha_A < 0:
+            raise ValueError("alpha_A must be positive")
+        if gamma >= 0:
+            raise ValueError("gamma must be negative")
+        if tol <= 0 or mtol <= 0 or eps_solver <= 0:
+            raise ValueError("tol and mtol must be positive")
+
+        self.w_evo = w_evo
+        self.threshold = threshold
+        self.eps_solver = eps_solver
+        self.relax_optim = relax_optim
+        self.m0 = m0
+        self.A0 = A0
+        self.thresholds = thresholds
+        self.alpha_A = alpha_A
+        self.alpha_m = alpha_m
+        self.eta = eta
+        self.gamma = gamma
+        self.PL = PL
+        self.PQ = PQ
+        self.tol = tol
+        self.mtol = mtol
+        self.accel = accel
+        self.thresholder = thresholder
+        self.reg = get_regularization(thresholder)
+        self.prox = get_prox(thresholder)
+        self.A_history_ = []
```

Comment:
Could you add these extra history objects to the list of attributes in the docstring?

Reply:
Doing this now